src/tests/ftest/soak/soak.py

```diff
@@ -65,239 +88,454 @@ def job_done(self, args):
+                status &= False
+                break
+            # Wait for rebuild to start
+            try:
+                pool.wait_for_rebuild(True)
+            except (RuntimeError, TestFail, DaosApiError) as error:
+                self.log.error(
+                    "Rebuild failed waiting to start", exc_info=error)
+                status &= False
+                break
+
+            # Wait for rebuild to complete
+            try:
+                pool.wait_for_rebuild(False)
+            except (RuntimeError, TestFail, DaosApiError) as error:
+                self.log.error(
+                    "Rebuild failed waiting to finish", exc_info=error)
+                status &= False
+                break
 
-    def create_ior_cmdline(self, job_params, job_spec, pool):
+        with H_LOCK:
+            self.harasser_results["REBUILD"] = status
+
+    def launch_snapshot(self):
+        """Create a basic snapshot of the reserved pool."""
+        self.log.info("<<Launch Snapshot>> at %s", time.ctime())
+        status = True
+        # Create container
+        container = TestContainer(self.pool[0])
+        container.namespace = "/run/container_reserved"
+        container.get_params(self)
+        container.create()
+        container.open()
+
+        obj_cls = self.params.get(
+            "object_class", '/run/container_reserved/*')
+
+        # write data to object
+        data_pattern = get_random_string(500)
+        datasize = len(data_pattern) + 1
+        dkey = "dkey"
+        akey = "akey"
+        obj, epoch = container.container.write_an_obj(
+            data_pattern, datasize, dkey, akey, obj_cls=obj_cls)
+        obj.close()
+        # Take a snapshot of the container
+        snapshot = DaosSnapshot(self.context)
+        try:
+            snapshot.create(container.container.coh, epoch)
+        except (RuntimeError, TestFail, DaosApiError) as error:
+            self.log.error("Snapshot failed", exc_info=error)
+            status &= False
+        if status:
+            self.log.info("Snapshot Created")
+            # write more data to object
+            data_pattern2 = get_random_string(500)
+            datasize2 = len(data_pattern2) + 1
+            dkey = "dkey"
+            akey = "akey"
+            obj2, _ = container.container.write_an_obj(
+                data_pattern2, datasize2, dkey, akey, obj_cls=obj_cls)
+            obj2.close()
+            self.log.info("Wrote additional data to container")
+            # open the snapshot and read the data
+            obj.open()
+            snap_handle = snapshot.open(
+                container.container.coh, snapshot.epoch)
+            try:
+                data_pattern3 = container.container.read_an_obj(
+                    datasize, dkey, akey, obj, snap_handle.value)
+            except (RuntimeError, TestFail, DaosApiError) as error:
+                self.log.error(
+                    "Error when retrieving the snapshot data.{0}".format(
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments